includes/repositories/EUtilsRepository.php

```diff
@@ -207,44 +198,84 @@ public function createAccession($accession) {
         'accession' => $accession['value'],
       ])->execute();
 
-      $this->insertDBXref($accession['value'], $db->db_id);
+      $this->createDBXref($accession['value'], $db->db_id);
 
       return static::$cache['accessions'][$accession['value']] = $this->getAccessionByID($id);
     }
 
     return NULL;
   }
 
-
   /**
    * Inserts tdbxref into the appropriate linker table, eg, project_dbxref.
    * dbxrefs are formatted db:accession.
    *
-   * @param $accession
+   * @param array $accession
+   * @param int $db_id
    *
-   * @param $db_id
+   * @return bool
+   * @throws \Exception
    */
-  private function insertDBXref($accession, $db_id) {
+  private function createDBXref($accession, $db_id) {
+    $this->validateBaseData();
 
     $dbxref = [
       'accession' => $accession,
       'db_id' => $db_id,
     ];
 
-    chado_associate_dbxref($this->base_table, $this->base_record_id, $dbxref);
-
+    return chado_associate_dbxref($this->base_table, $this->base_record_id,
+      $dbxref);
   }
 
   /**
    * Associates the XML with the record via the local:full_ncbi_xml term.
-   * @param $xml
-   * xml string as returned by simpleXML.
+   *
+   * @param string $xml string as returned by SimpleXMLElement.
    */
   public function createXMLProp($xml) {
-
     $xml_term = tripal_get_cvterm(['id' => 'local:full_ncbi_xml']);
 
-    $this->createProperty($xml_term->cvterm_id, $xml);
+    return $this->createProperty($xml_term->cvterm_id, $xml);
+  }
+
+  /**
+   * Set the record id.
+   *
+   * @param int $id
+   *
+   * @return $this
+   */
+  public function setBaseRecordId($id) {
+    $this->base_record_id = $id;
+    return $this;
+  }
+
+  /**
+   * Set the base table.
```

Comment:
Adding Chado here since i think this can be confusing...

```suggestion
   * Set the Chado base table.
```